test/connect-backbone-to-react.test.js

```diff
@@ -594,6 +594,76 @@ describe('connectBackboneToReact', function() {
     });
   });
 
+  describe('when passed props change to include', function() {
+    let ConnectedTest;
+    let setStateSpy;
+    let createListenerSpy;
+    let removeListenerSpy;
+
+    beforeEach(function() {
+      ConnectedTest = connectBackboneToReact(mapModelsToProps)(TestComponent);
+      wrapper = mount(<ConnectedTest models={modelsMap} />);
+
+      setStateSpy = sandbox.spy(ConnectedTest.prototype, 'setState');
+      createListenerSpy = sandbox.spy(ConnectedTest.prototype, 'createEventListener');
+      removeListenerSpy = sandbox.spy(ConnectedTest.prototype, 'removeEventListener');
+
+      const decoratorUserModel = new UserModel({
+        name: 'Robert',
+        age: '30',
+        hungry: false,
+      });
+
+      const initialModelsMap = {
+        user: userModel,
+        coll: userCollection,
+        decorator: decoratorUserModel,
+      };
+
+      wrapper.setProps({ models: initialModelsMap });
+    });
+
+    afterEach(function() {
+      wrapper.unmount();
+    });
+
+    it('calls setState once', function() {
+      assert.equal(setStateSpy.callCount, 1);
+    });
+
+    it('calls createEventListener once for the model', function() {
+      assert.equal(createListenerSpy.callCount, 1);
```

Comment:
How is this only called once? Don't we have 3 Models/Collections? Shouldn't this be called 3 times? I'm a little confused.

Reply:
It's only called once because the userModel and userCollection haven't changed between the mounting of the view [here](https://github.com/mongodb-js/connect-backbone-to-react/pull/21/files#diff-f93d5182925707305633f3d1d9f6a73bR605) and the props update [here](https://github.com/mongodb-js/connect-backbone-to-react/pull/21/files#diff-f93d5182925707305633f3d1d9f6a73bR623). Since only the decorator has changed since the listeners were set up, createEventListener has only been called for the decorator.

Reply:
`calls createEventListener once due to decoratorUserModel being added as a model`?